autoPyTorch/evaluation/tae.py

```diff
@@ -182,6 +169,33 @@ def __init__(
 
         self.search_space_updates = search_space_updates
 
+        if isinstance(self.resampling_strategy, (HoldoutValTypes, CrossValTypes)):
+            eval_function = autoPyTorch.evaluation.train_evaluator.eval_function
+        elif isinstance(self.resampling_strategy, NoResamplingStrategyTypes):
+            eval_function = autoPyTorch.evaluation.fit_evaluator.eval_function
```

Comment:
Can we merge `fit_evaluator` and `train_evaluator` and process NoResamplingStrategyTypes separately?
The names are confusing and not clear.